store/src/java/com/zimbra/cs/mailbox/Notification.java

```diff
@@ -523,6 +523,10 @@ private void notifyIfNecessary(Account account, Message msg, String rcpt)
                 ZimbraLog.mailbox.warn("error encoutered looking up return path configuration, using null return path instead", se);
             }
             smtpSession.getProperties().setProperty("mail.smtp.from", envFrom);
+            
+            MailSender.setupStartTlsMode(account, smtpSession);
+            ZimbraLog.smtp.debug("Sending message %s to SMTP host %s with properties: %s",
+            		mm.getMessageID(), smtpSession.getProperties().getProperty("mail.smtp.host"), smtpSession.getProperties());
```

Comment:
There are several other places where the Transport.send() is called.  For example, in the  com.zimbra.cs.mailbox.Notification.interceptIfNecessary(). Don't we need to take care of those calls as well?

Reply:
Definitely looks like we need to cover those. Maybe we can find an easier way to fit this into the existing call flows without touching each call point?

Reply:
Ok, this is bigger animal than I expected.  There are total 11 references for Transport.send() or Transport.sendMessage() in zm-mailbox, excluding test code and smtpInject tool.
I have updated 2 of them .
How about to unify the all smtp submissions into single method in MailSender class?
(Transfer.send proxy)

Reply:
Transport.send (and transport.sendMessage ) takes MimeMessage as argument.
MimeMessage is created with Session.
And the Session is created by JMSession.get*Session methods.
In these methods, JavaMail SMTP properties (host, port, timeout, etc) are configured.
- JMSession.getSmtpSession(account)
  - This method doesn't configure STARTTLS.
  - I will move the starttls related config logic from MailSender to this method.
- JMSession.getRelaySession()
  - This method refers "zimbraShareNotificationMtaConnectionType" (CLEARTEXT,SSL, or STARTTLS).
  - This method is used on connecting zimbraShareNotificationMtaHostname host to send share notification. If the attr is not defined, it will fallback to getSmtpSession(null).
- JMSession.getSession(DataSource)
  - This method is used on connecting smtp host defined in the datasource.
  - This method refers local config "javamail_smtp_enable_starttls". There are ldap attributes named "zimbraDataSourceConnectionType", but it is used only for POP and IMAP.
  - Should I add the logic here also?